File: src/chipschallenge/buttonbehaviors/NullButtonBehavior.java
package chipschallenge.buttonbehaviors;

import chipschallenge.Block;

/**
 *
 * @author patrik
 */
public class NullButtonBehavior extends ButtonBehavior {

    private NullButtonBehavior() {
    }
    private static NullButtonBehavior mInstance = null;

    public static synchronized NullButtonBehavior getInstance() {
        if (mInstance == null) {
            mInstance = new NullButtonBehavior();
        }
        return mInstance;
    }

    public void buttonDown(Block moving, Block button) {
        // Do nothing
    }

    public void buttonUp(Block listener, Block button) {
        // Do nothing
    }
}

File: ChipsChallenge.Shared/Buttonbehaviors/NullButtonBehavior.cs
namespace ChipsChallenge.Shared.Buttonbehaviors
{
    public class NullButtonBehavior : IButtonBehavior
    {
        private static NullButtonBehavior instance;

        private NullButtonBehavior()
        {
        }

        public static NullButtonBehavior Instance
        {
            get
            {
                lock (typeof(NullButtonBehavior))
                {
                    return instance ?? (instance = new NullButtonBehavior());
                }
            }
        }

        public virtual void ButtonDown(Block moving, Block button)
        {
        }

        public virtual void ButtonUp(Block listener, Block button)
        {
        }
    }
}